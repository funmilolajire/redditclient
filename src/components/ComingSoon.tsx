import { useRef, useLayoutEffect } from 'react';
import { gsap } from 'gsap';

export const ComingSoon = () => {
    const boxRef = useRef<HTMLDivElement>(null);

    useLayoutEffect(() => {
        const tl = gsap.timeline({ repeatDelay: 2 })
        tl.from([boxRef.current], {
            y: '-55vh',
            duration: 3,
            delay: 2,
            opacity: 0,
            ease: 'elastic'
        })
    })

    return (
        <div ref={boxRef} className="container flex items-center justify-center h-screen font-extrabold ComingSoon">
            <div className="bg-yellow-500 border-2 border-gray-800 border-solid skew-y-2 transform-gpu text-gray-50 text-9xl">
                <p>Coming Soon</p>
            </div>
        </div>
    )
}