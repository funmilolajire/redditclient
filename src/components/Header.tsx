import { useRef, useLayoutEffect } from 'react';
import { gsap } from 'gsap';
import logo from '../images/logo192.png';

export const Header = () => {
    const boxRef = useRef<HTMLDivElement>(null);
    const logoRef = useRef<HTMLImageElement>(null);
    const titleRef = useRef<HTMLHeadingElement>(null);

    useLayoutEffect(() => {
        const tl = gsap.timeline({});
        tl.from([boxRef.current], {
            duration: 1
        })
        tl.from([logoRef.current], {
            x: '-3000px',
            opacity: 0,
            ease: 'circ'
        })
        tl.from([titleRef.current], {
            x: '3000px',
            opacity: 0,
            ease: 'slow'
        })
    })

    return (
        <header>
            <div className="header-inner">
                <section ref={boxRef}>
                    <figure ref={logoRef}>
                        <img src={logo} alt="site logo" />
                    </figure>
                    <h1 ref={titleRef}>
                        <span>Reddit</span>Client
                    </h1>
                </section>
            </div>
        </header>
    )
}