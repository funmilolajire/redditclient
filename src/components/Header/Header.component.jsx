import logo from './logo192.png';
import { useRef, useEffect } from 'react';
import { gsap } from 'gsap';

export const Header = () => {
    const boxRef = useRef();
    const logoRef = useRef();
    const titleRef = useRef();

    useEffect(() => {
        const tl = gsap.timeline({});
        tl.from([boxRef.current], {
            duration: 1
        })
        tl.from([logoRef.current], {
            x: '-3000px',
            opacity: 0,
            ease: 'circ'
        })
        tl.from([titleRef.current], {
            x: '3000px',
            opacity: 0,
            ease: 'slow'
        })
    })

    return (
        <div className="relative z-10 py-4 bg-gray-100 shadow-2xl">
            <section ref={boxRef} className="flex items-center justify-center">
                <figure ref={logoRef} className="w-20">
                    <img className="w-full h-auto" src={logo} alt="site logo" />
                </figure>
                <h1 ref={titleRef} className="pl-5 text-6xl" style={{ fontFamily: "Stylish, sans-serif" }}><span className="text-yellow-500">Reddit</span>Client</h1>
            </section>
        </div>
    )
}