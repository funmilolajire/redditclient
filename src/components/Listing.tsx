import { useRef, useLayoutEffect, useCallback } from 'react';
import { useAppDispatch, useAppSelector } from '../store/hooks';
import { handleListingChange, handleTermChange } from '../store/slices/postsSlice';

export const Listing = ({ listItem }: { listItem: string }) => {
    const listingRef = useRef<HTMLLIElement>(null);

    //handle listing change and active class
    const dispatch = useAppDispatch();
    const listing = useAppSelector(state => state.posts.listing);
    const handleListing = () => {
        if (null !== listingRef.current && listing !== listingRef.current.id) {
            let newListing = listingRef.current.id;
            dispatch(handleListingChange(newListing))
            dispatch(handleTermChange(''))
        }
    }
    const handleActiveClass = useCallback(() => {
        if (null !== listingRef.current) {
            const classList = listing === listingRef.current.id ? "Listing active" : "Listing";
            listingRef.current.className = classList
        }
    }, [listing])
    useLayoutEffect(() => {
        handleActiveClass();
    }, [handleActiveClass])


    //JSX
    return (
        <li ref={listingRef} data-testid={listItem} onClick={handleListing} id={listItem}>
            {listItem}
        </li>
    )
}