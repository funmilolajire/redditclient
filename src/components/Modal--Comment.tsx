import { useRef } from 'react';
import parse from 'html-react-parser';
import { BiComment } from 'react-icons/bi';
import { BsDot } from 'react-icons/bs';
import { GoArrowDown, GoArrowUp } from 'react-icons/go';

interface ModalCommentType {
    [key: string]: any
}

export const ModalComment = ({ comment }: { comment: ModalCommentType }) => {
    const modalCommentRef = useRef<HTMLDivElement>(null);

    //parse comment html
    const parsedHTML = (html: string) => html ? parse(parse(html).toString()) : null;

    //JSX Element
    return (
        <div className="ModalComment" ref={modalCommentRef} data-permalink={comment.permalink}>
            <p className="Comment--Info">comment made <span className="time">{comment.time}</span> by <span className="author">{comment.author}</span></p>
            <div className="Comment--Body">
                {parsedHTML(comment.comment)}
            </div>
            <div className="Comment--Bottom">
                <div className="Comment--Replies">
                    <BiComment className="reply-icon icon" />
                    <span className="comments value">{comment.num_replies}</span>
                    <span className="label">{comment.num_replies !== 1 ? 'replies' : 'reply'}</span>
                </div>
                <BsDot className="dot-icon" />
                <div className="Votes Comment--Upvotes">
                    <GoArrowUp className="arrow-icon icon" />
                    <span className="upvotes value">{comment.upvotes}</span>
                    <span className="label">{comment.upvotes !== 1 ? 'upvotes' : 'upvote'}</span>
                </div>
                <BsDot className="dot-icon" />
                <div className="Votes Comment--Downvotes">
                    <GoArrowDown className="arrow-icon icon" />
                    <span className="downvotes value">{comment.downvotes}</span>
                    <span className="label">{comment.downvotes !== 1 ? 'downvotes' : 'downvote'}</span>
                </div>
            </div>
        </div>
    )
}