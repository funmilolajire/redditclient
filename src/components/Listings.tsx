import { useRef, useLayoutEffect } from 'react';
import { Listing } from './Listing';
import { gsap } from 'gsap';

export const Listings = () => {
    const listings: string[] = ['hot', 'new', 'top', 'best', 'rising', 'controversial'];

    //animations
    const listingBoxRef = useRef<HTMLDivElement>(null);
    useLayoutEffect(() => {
        const tl = gsap.timeline({})
        tl.from([listingBoxRef.current], {
            x: '1200%',
            delay: 2,
            duration: 2,
            ease: 'power3.inOut'
        })
    }, [])

    return (
        <div ref={listingBoxRef} className="Listings">
            <ul className="Listings--List">
                {
                    listings && listings.map(listing => <Listing key={listing} listItem={listing} />)
                }
            </ul>
        </div>
    )
}