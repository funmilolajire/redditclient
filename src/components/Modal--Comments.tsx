import { ModalComment } from "./Modal--Comment";

interface Comment {
    [key: string]: any
}
interface ModalCommentsType {
    comments: Comment[]
}

export const ModalComments = (props: ModalCommentsType) => {
    const { comments } = props;
    return (
        <div className="Modal--Comments">
            {
                comments.length !== 0 ?
                    <>
                        <h3>Comments</h3>
                        <div className="comments">
                            {comments && comments.map((comment: Comment) => <ModalComment key={comment.id} comment={comment} />)}
                        </div>
                    </> :
                    <p className="no--comments">No comments yet</p>
            }
        </div>
    )
}