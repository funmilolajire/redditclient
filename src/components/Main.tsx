import { Listings } from './Listings';
import { Posts } from './Posts';
export const Main = () => (
    <div className="Main">
        <Listings />
        <Posts />
    </div>
)