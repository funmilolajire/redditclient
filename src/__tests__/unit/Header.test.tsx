import { mount, shallow } from 'enzyme';
import { Header } from '../../components/Header';

describe('renders Header component', () => {
    const HeaderComponent = mount(<Header />)

    describe('make snapshot of Header component', () => {
        test('matches the snapshot', () => {
            const component = shallow(<Header />);
            expect(component).toMatchSnapshot();
        })
    })
    describe('Header Content', () => {
        const getText = HeaderComponent.find('h1').text()
        const getLogo = HeaderComponent.find('img')
        test('Contains Site Title', () => {
            expect(getText).toEqual('RedditClient');
        })
        test('Site Title is string', () => {
            expect(getText).toBeString();
        })
        test('Contains Site Logo', () => {
            expect(getLogo).toHaveLength(1)
        })
    })
})