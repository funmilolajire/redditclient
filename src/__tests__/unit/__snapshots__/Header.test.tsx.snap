// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Header component make snapshot of Header component matches the snapshot 1`] = `
<header>
  <div
    className="header-inner"
  >
    <section>
      <figure>
        <img
          alt="site logo"
          src="logo192.png"
        />
      </figure>
      <h1>
        <span>
          Reddit
        </span>
        Client
      </h1>
    </section>
  </div>
</header>
`;
